### PR TITLE
Implement new burn-in flow

### LINKED ISSUE #240: Redesigned burn-in workflow
Instead of starting a CI job from processbot, the burn-in workflow should work as follows:

* developer adds the label `A1-needsburnin` on a PR
* processbot posts a comment to the PR with a default `request` file and an explanation how to submit that request
* developer submits a comment with `bot burnin` and the content of the to-be-created `request` file
* processbot checks that developer is authorized to submit burn-in requests (i.e. is a member of `paritytech` org)
* processbot commits a new `request` file to the Gitlab project for managing burn-in tests and responds with a confirmation comment on the PR

### PR DESCRIPTION
**Pull request by haikoschol**
Processbot will now react to the `A1-needsburnin` label being added with a comment that explains how to submit a burn-in request.

When such a request is received via PR comment by the developer, the included ToML content is committed to the Gitlab repository used for managing burn-in tests.

closes #240

### CODE REVIEW

File: src/webhook.rs [hunk 1]
```diff
@@ -635,113 +663,135 @@ async fn handle_comment(
 				number,
 			))))?;
 		}
-	} else if body.to_lowercase().trim() == BURNIN_REQUEST.to_lowercase().trim()
+	} else if body
+		.to_lowercase()
+		.trim()
+		.starts_with(BURNIN_REQUEST.to_lowercase().trim())
```
Comment (Xanewok) [inline src/webhook.rs:669]: I don't know if the commands are known to be ASCII, but if that's the case then we might want to simplify that by calling `str::eq_ignore_ascii_case` on the `trim`med strings

**Review by Xanewok — COMMENTED** [review]
I'm not so well versed with the ops side of things so can't comment on the logic changes; otherwise looks good, left some nits

**Comment by joao-paulo-parity** [comment]
I did not forget about this PR. Before I switched to working on some other projects, I wanted to also get the [integration test branch](https://github.com/paritytech/parity-processbot/tree/jp_companion) merged in but it's difficult to reason about implementing more features when the code is not in a good state.

I'll begin an effort to refactor the code base (https://github.com/paritytech/parity-processbot/issues/293) and then I could adapt this PR after that's done.

**Comment by haikoschol** [comment]
My bad for letting this PR rot. Also, @Xanewok thanks for the review and sorry for not following up on it. Like @joao-paulo-parity, I was caught up in other projects (or rather still am).

> I'll begin an effort to refactor the code base (#293) and then I could adapt this PR after that's done.

Don't bother working on this PR. In fact I'll close it now. The approach here is overengineered and opens potential security issues with having to keep auth tokens for Gitlab around. Plus another failure mode without good error handling when the request to Gitlab fails.

Now that the frontend for submitting burn-in requests is finally accessible to devs, the bot should simply reply with a link to it and a short explanation of the process. I will get to implementing that eventually. :)

**Pull request closed without merging**